### PR TITLE
Fix element selection when same selector is used in different targets

### LINKED ISSUE #10: Returns the wrong element when calling it with the same selector, but in different targets
I came across this scenario:

```html
<div id="target1">
  <p class="unicorn">Some element</p>
</div>

<div id="target2">
  <span class="unicorn">Some other completely different element</span>
</div>
```

Using `elementReady` in the following way with this markup will output:

```js
// Returns the <p> element (correct)
elementReady('.unicorn', { target: $('#target1') }); 

// Returns the same <p> element (wrong), should return <span>
elementReady('.unicorn', { target: $('#target2') }); 
```

The culprit is the caching implementation, which is not taking into account the targets, just the selectors.

### PR DESCRIPTION
**Pull request by reyronald**
In the previous behavior, `elementReady` returns the wrong element when calling it with the same selector, but in different targets

Fixes #10 

----

1. The caching implementation got a little convoluted because of the nature of the problem. If anyone has any suggestions to make it more readable let me know!
2. Is there any edge-case that I didn't consider?
3. Does anyone think it's necessary to add a more complex test?
4. Also, shouldn't we clean the cache in the `onCancel` event as well?

Pinging @bfred-it , who also worked in this

### CODE REVIEW

**Comment by bfred-it** [comment]
@sindresorhus do you know of a multi-key map? In this case:

```js
const cache = new MultiMap();
cache.set(target, selector, promise);
cache.get(target, selector);
cache.has(target, selector);
```

**Comment by reyronald** [comment]
That MultiMap sounds like a good idea for a standalone package, if it doesn't exist already.

**Comment by bfred-it** [comment]
Something like: https://runkit.com/bfred-it/multimap-demo

Notes:
- almost too flexible, `.get(target)` would return `Map([selector, promise])`
- not sure I want to complete/test/publish this 😅

**Comment by sindresorhus** [comment]
> Also, shouldn't we clean the cache in the onCancel event as well?

Yes, we should. Can you do it?

**Comment by sindresorhus** [comment]
Other than the above, it looks good :)

**Comment by reyronald** [comment]
> Yes, we should. Can you do it?

Sure, no problem. I'll do it a little later today.

I guess the MultiMap suggestion can be left for another improvement PR in the future then?

**Comment by sindresorhus** [comment]
> I guess the MultiMap suggestion can be left for another improvement PR in the future then?

Yup

**Comment by reyronald** [comment]
There we ago, I included a small test too. Lemme know if you pick on anything else. :)

File: index.js [hunk 1]
```diff
@@ -1,21 +1,29 @@
 'use strict';
 const PCancelable = require('p-cancelable');
 
-const selectorCache = new Map();
+const targetCache = new Map();
```
Comment (bfred-it) [inline index.js]: This also works as a `WeakMap`

Comment (reyronald) [reply index.js]: You are right! That wasn't the case before though, the only map that existed initially is now the inner map of that `targetCache`, which still has to remain as a regular `Map` .

Comment (bfred-it) [reply index.js]: No worries, I've just been trying to look for places where `WeakMap` can work :)

**Comment by sindresorhus** [comment]
Thanks for fixing 🙌

**Comment by reyronald** [comment]
My pleasure! 😄
